### PR TITLE
Improved makefile in some areas

### PR DESCRIPTION
**Pull request by andy-maier**
For details, see the commit message.
Ready for review and merge.

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -355,7 +374,7 @@ clobber: clean
 clean:
 	@echo "makefile: Removing temporary build products"
 	find . -name "*.pyc" -delete
-	sh -c "find . -name \"__pycache__\" |xargs rm -Rf __pycache__"
+	bash -c "find . -name \"__pycache__\" |xargs -n 1 rm -Rf"
```
Comment (KSchopmeyer) [inline makefile]: Do we really need bash.  There may be some platforms left over that still have sh as the default shell.

File: makefile [hunk 2]
```diff
@@ -517,7 +536,7 @@ ifeq ($(PLATFORM),Windows)
 else
 	testsuite/test_uprint.sh
 endif
-	bash -c "set -o pipefail; PYTHONWARNINGS=default py.test --cov $(package_name) --cov $(mock_package_name) $(coverage_report) --cov-config coveragerc --ignore=attic --ignore=releases -s 2>&1 |tee $(test_tmp_file)"
+	bash -c "set -o pipefail; PYTHONWARNINGS=default py.test --cov $(package_name) --cov $(mock_package_name) $(coverage_report) --cov-config coveragerc $(pytest_opts) --ignore=attic --ignore=releases -s 2>&1 |tee $(test_tmp_file)"
```
Comment (KSchopmeyer) [inline makefile]: Same comment

File: makefile (hunk 1 above)
Comment (andy-maier) [reply makefile]: We do depend on bash features in several of our .sh scripts, as well as in several places in makefiles (if conditions, pipefail, ...).

bash is the standard shell in Linux. OS-X has it, too. CygWin has it. There is more danger of not having Bourne shell on Linux than not having bash. If you know of any Linux platforms that don't have bash, post them here.

I checked and I did not find any other sh holdouts in pywbem or pywbemtools. If you know of any, post them here.

Comment (KSchopmeyer) [reply makefile]: Sounds good to me.  Forget my comment.

**Comment by andy-maier** [comment]
Karl agreed in slack.

**Pull request merged**